pkg/runner/builtin/runner.go

```diff
@@ -283,11 +235,20 @@ func cname(job *types.Job) string {
 
 func (r *Runner) Poll(q queue.Queue) {
 	for {
-		w := q.Pull()
-		q.Ack(w)
-		err := r.Run(w)
-		if err != nil {
-			log.Error(err)
+		stats := r.Manager.ClusterStats()
+		log.Errorf("Cluster stats:")
+		log.Errorf("CPU: %s", stats.Cpus)
+		log.Errorf("Reserved CPU: %s", stats.ReservedCpus)
+		// If available cpus
+		if stats.Cpus - stats.ReservedCpus > 0 {
+			w := q.Pull()
+			q.Ack(w)
+			log.Errorf("Poll queue")
+			go r.Run(w)
```

Comment:
I'll need to dig deeper, but we should make sure that this function can recover in case of a panic, typically done with `defer recover()`.